test/aws_test.go

```diff
@@ -4,32 +4,244 @@
+			envVar:  "PART_SIZE",
+			value:   "", //TODO
+		},
+		/*{
+			helmKey: "pachd.storage.amazon.retries",
+			envVar:  "RETRIES",
+			value:   "", //TODO
+		},*/
+		/*{
+			helmKey: "pachd.storage.amazon.reverse",
+			envVar:  "REVERSE",
+			value:   "", //TODO
+		},*/
+		{
+			helmKey: "pachd.storage.amazon.timeout",
+			envVar:  "TIMEOUT",
+			value:   "", //TODO
+		},
+		{
+			helmKey: "pachd.storage.amazon.uploadACL",
+			envVar:  "UPLOAD_ACL",
+			value:   "", //TODO
+		},
+	}
+	var (
+		expectedServiceAccount = "my-fine-sa"
+		expectedProvisioner    = "ebs.csi.aws.com"
+		expectedStorageBackend = "AMAZON"
+	)
+
+	helmValues := map[string]string{
+		"pachd.storage.backend": "AMAZON",
+		`pachd.serviceAccount.additionalAnnotations.eks\.amazonaws\.com/role-arn`:  expectedServiceAccount,
+		`worker.serviceAccount.additionalAnnotations.eks\.amazonaws\.com/role-arn`: expectedServiceAccount,
+	}
+	for _, tc := range testCases {
+		helmValues[tc.helmKey] = tc.value
+	}
+
+	templatesToCheck := map[string]bool{
+		"templates/pachd/storage-secret.yaml":             false,
+		"templates/pachd/deployment.yaml":                 false,
+		"templates/pachd/rbac/serviceaccount.yaml":        false,
+		"templates/pachd/rbac/worker-serviceaccount.yaml": false,
+		"templates/etcd/statefulset.yaml":                 false,
+		"templates/etcd/storageclass-aws.yaml":            false,
+		"templates/postgresql/statefulset.yaml":           false,
+		"templates/postgresql/storageclass-aws.yaml":      false,
+	}
+
+	templatesToRender := []string{}
+	for k := range templatesToCheck {
+		templatesToRender = append(templatesToRender, k)
+	}
+
+	objects, err := manifestToObjects(helm.RenderTemplate(t, &helm.Options{
+		SetValues: helmValues,
+	}, "../pachyderm", "blah", templatesToRender))
+
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
-	for _, obj := range objects {
-		switch obj := obj.(type) {
+
+	for _, object := range objects {
+		switch resource := object.(type) {
 		case *v1.Secret:
-			if obj.Name != "pachyderm-storage-secret" {
+			if resource.Name != "pachyderm-storage-secret" {
 				continue
 			}
-			if _, err := strconv.Atoi(string(obj.Data["part-size"])); err != nil {
-				t.Error(err)
+
+			for _, tc := range testCases {
+				t.Run(fmt.Sprintf("%s equals %s", tc.envVar, tc.value), func(t *testing.T) {
+					if got := string(resource.Data[tc.envVar]); got != tc.value {
+						t.Errorf("got %s; want %s", got, tc.value)
+					}
+				})
+			}
+			templatesToCheck["templates/pachd/storage-secret.yaml"] = true
+
+		case *v1.ServiceAccount:
+			if resource.Name == "pachyderm-worker" || resource.Name == "pachyderm" {
+
+				t.Run(fmt.Sprintf("%s service account annotation equals %s", resource.Name, expectedServiceAccount), func(t *testing.T) {
+					if sa := resource.Annotations["eks.amazonaws.com/role-arn"]; sa != expectedServiceAccount {
+						t.Errorf("expected service account to be %q but was %q", expectedServiceAccount, sa)
```

Comment:
The output of the test should always be got, want, and they should be clearly identified: `service account:  got "foo", want "bar"`.  I like to put got and want on separate lines, indent, and align the colons, but you don't have to do that. 

https://github.com/golang/go/wiki/TestComments#got-before-want

Reply:
Will address in follow on PR